### ISSUE TITLE
[Bug] Crypto report breaks the terminal

### ISSUE THREAD

**Issue by jose-donato** [GamestonkTerminal/GamestonkTerminal#874]
**Describe the bug**
Crypto report (currently number 4) breaks the application

**To Reproduce**
1. `reports`
2. `4`

**Screenshots**
![image](https://user-images.githubusercontent.com/43375532/139274958-3b905a01-3142-48a9-b7c7-f3649aae8bf8.png)

**Desktop (please complete the following information):**
 - OS: MacOS Big Sur 11.6
 - Python version: Python 3.8.12

**Additional context**
Coingecko API was up when I ran the report command

**Comment by piiq** [comment]
> We probably want to add retries.

We do have 3 default retries with a 1 sec `backoff_factor` when the persistent session is set up. See [here](https://github.com/GamestonkTerminal/GamestonkTerminal/blob/28e3970912b3f81b59ca2de5d77e6034c7e581c1/gamestonk_terminal/cryptocurrency/pycoingecko_helpers.py#L40)

We can increase the retry count to 5 which with out `backoff_factor` should give more "air" for the requests to handle some unstable or bottleneck connections.

Here's an extract from the urllib3 docs about the backoff factor to understand the context:


> A backoff factor to apply between attempts after the second try (most errors are resolved immediately by a second try without a delay). urllib3 will sleep for: `{backoff factor} * (2 ** ({number of total retries} - 1))`
> seconds. If the backoff_factor is 0.1, then sleep() will sleep for [0.0s, 0.2s, 0.4s, …] between retries. It will never be longer than Retry.BACKOFF_MAX.

**Comment by aia** [comment]
Coingecko had issues all week for me even in their own mobile app.

**Comment by piiq** [comment]
And now we have the same error in the tests. See https://github.com/GamestonkTerminal/GamestonkTerminal/runs/4116994898?check_suite_focus=true

**Comment by piiq** [comment]
An update on this issues.
In the big UX branch we've added exception handling so these errors will not break the terminal.

While I was implementing this exception handling I noticed that the coingecko scrapper is [fubar](https://en.wikipedia.org/wiki/FUBAR_(disambiguation)). Coingecko have and API that has a free tier that is slightly more generous that the free tier of alphavantage (that's enough) so if we don't have specific reason to parse their website we should be transitioning to the usage of the API.
